lib/net/x86-64-linux/Net.v3

```diff
@@ -0,0 +1,267 @@
+// Copyright 2022 Kunal Sareen. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+// Implementation of the "Net" component for Linux, using underlying Linux
+// syscalls and methods.
+
+// XXX: Unsound as we need to enforce that ipv4 and ipv6 are exactly 4 and 16
+// u8s long respectively. One way to achieve this could be to create two more
+// types, Inet4Array and Inet6Array, like so:
+// class Inet4Array {
+// 	def addr: Array<u8> = Array.new(4);
+// }
+// class Inet6Array {
+// 	def addr: Array<u8> = Array.new(16);
+// }
+// type InetAddress {
+// 	case Inet4(addr: Inet4Array);
+// 	case Inet6(addr: Inet6Array);
```

Comment:
I think it's best to leave Inet6 out until it's fully implemented.

Reply:
Sure. Is it fine if I just comment it out instead of deleting it? There's some substantial stuff for IPv6 like the rendering code.

Reply:
PS do you have a suggestion on how to deal with array sizes for the two kinds of IP addresses? Ideally the arrays are only 4- and 16-bytes long for IPv4 and IPv6 respectively. Currently it's just `Array<u8>` which can be any array length.

Reply:
I think it's best not to have commented-out code checked in. I only very rarely do. I keep experimental code in a separate directory or as a sketch outside the repo.

Reply:
For the array sizes, what about this?

```
type INetAddress {
    case IPV4(oct3: u8, oct2: u8, oct1: u8, oct0: u8) { }
    case IPV6(word3: u32, word2: u32, word1: u32, word9: u32) { }

    def put(r: Range<u8>) {
        match (this) {
           case IPV4(b3, b2, b1, b0) => { r[0] = b3; r[1] = b2; r[2] = b1; r[2] = b1; r[3] = b0; }
           case IPV6 ...
        }
    }
```

That way, `INetAddress` values will always compare equal, and there will be no array object overhead.

Reply:
It works for IPv4 but I feel like it'll make some operations for IPv6 a bit painful such as the rendering code. It's nicer to work with bytes directly instead of having to pack and unpack them into a `u32`.

> Also, using a Range<T> allows writing the types out without relying on DataReader.

Not sure I follow. I'm not using a `DataReader` anywhere in the code, just directly copying to/from the C-struct definition.

Reply:
I added IPv6 support -- it was surprisingly easy to add. I'll think about how to fix up the array size issue.